packages/pwa/app/commerce-api/hooks/useCustomerProductList.js

```diff
@@ -0,0 +1,325 @@
+            },
+
+            get isEmpty() {
+                return !self.data?.customerProductListItems?.length
+            },
+
+            findItemByProductId(productId) {
+                return self.data?.customerProductListItems?.find(
+                    (item) => item.productId === productId
+                )
+            },
+
+            isProductInList(productId) {
+                return !!self.findItemByProductId(productId)
+            },
+
+            /**
+             * Initialize customer's product list.
+             */
+            async init() {
+                actions.setInitialized(false)
+                const list = await self._getOrCreatelistByName(name, {type})
+                const productDetails = await self._getProductDetails(list)
+                const result = self._mergeProductDetailsIntoList(list, productDetails)
+                actions.receiveList(result)
+                actions.setInitialized(true)
+                return list
+            },
+
+            async getList(listId) {
+                const response = await api.shopperCustomers.getCustomerProductList({
+                    parameters: {
+                        customerId: customer.customerId,
+                        listId
+                    }
+                })
+
+                if (isError(response)) {
+                    throw new Error(response)
+                }
+
+                return response
+            },
+
+            async addItem(product) {
+                let list = self.data
+                if (!self.isInitialized) {
+                    list = await self.init()
+                }
+                const createdItem = await self.createListItem(list.id, product)
+                actions.createListItem(list.id, createdItem)
+            },
+
+            async updateItem(item) {
+                const {id, quantity} = item
+                if (quantity === 0) {
+                    await self.removeListItem(self.data.id, id)
+                    actions.removeListItem(self.data.id, id)
+                    return
+                }
+                const updatedItem = await self.updateListItem(self.data.id, item)
+                actions.updateListItem(self.data.id, updatedItem)
+            },
+
+            async removeItem(itemId) {
+                await self.removeListItem(self.data.id, itemId)
+                actions.removeListItem(self.data.id, itemId)
+            },
+
+            async removeItemByProductId(productId) {
+                const item = self.findItemByProductId(productId)
+                if (!item) {
+                    return
+                }
+                await self.removeItem(item.id)
+            },
+
+            /**
+             * Adds an item to the customer's product list.
+             * @param {object} listId
+             * @param {Object} item item to be added to the list.
+             */
+            async createListItem(listId, item) {
```

Comment:
Good rule of thumb would be: "If the function signature has `listId` in it, then it should be in the useCustomerProductLists (plural) hook". 

I'd also consider splitting up these 2 hooks to have their own files. To reduce confusion.